File: amortization-table.client/src/shared/models/AmortizationRequest.ts
export class AmortizationRequest {

  public interest: number | undefined;
  public purchasePrice: number | undefined;
  public downPayment: number | undefined;
  public term: number | undefined;
  public paymentFrequency: number | undefined;
  public termType?: 'Week' | 'Month' | 'Year';
  public loanStartDate?: string | null;

}

File: amortization-table.Server/Models/AmortizationRequest.cs
namespace amortization_table.Server.Models
{
    public class AmortizationRequest
    {
        public double interest { get; set; }
        public double purchasePrice { get; set; }
        public double downPayment { get; set; }
        public int? term { get; set; }
        public double? paymentFrequency { get; set; }
        public string? termType { get; set; }
        public DateTime? loanStartDate { get; set; }
        public AmortizationRequest() { }

    }
}
